Sources/App/Commands/Reconcile.swift

```diff
@@ -117,5 +117,6 @@ func reconcileLists(db: Database, source: [URL], target: [URL]) async throws {
 }
 
 func processPackageDenyList(packageList: [URL], denyList: [URL]) -> [URL] {
-    return Array(Set(packageList).subtracting(Set(denyList)))
+    let lowercaseDenyList = denyList.map { $0.absoluteString.lowercased() }
+    return Array(packageList.filter { lowercaseDenyList.contains($0.absoluteString.lowercased()) == false })
```

Comment:
It'd be better to use case-insensitive hashing to avoid the quadratic filter:

```swift
func processPackageDenyList(packageList: [URL], denyList: [URL]) -> [URL] {
    struct CaseInsensitiveURL: Equatable, Hashable {
        var url: URL

        init(_ url: URL) {
            self.url = url
        }

        func hash(into hasher: inout Hasher) {
            hasher.combine(url.absoluteString.lowercased())
        }

        static func == (lhs: Self, rhs: Self) -> Bool {
            lhs.url.absoluteString.lowercased() == rhs.url.absoluteString.lowercased()
        }
    }

    return Array(
        Set(packageList.map(CaseInsensitiveURL.init))
            .subtracting(Set(denyList.map(CaseInsensitiveURL.init)))
    ).map(\.url)
}
```

Reply:
Integrated this. Thanks!